src/main/java/com/epam/ta/reportportal/dao/ActivityRepositoryCustomImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.ta.reportportal.dao;
+
+import com.epam.ta.reportportal.commons.querygen.Condition;
+import com.epam.ta.reportportal.commons.querygen.Filter;
+import com.epam.ta.reportportal.commons.querygen.FilterCondition;
+import com.epam.ta.reportportal.commons.querygen.QueryBuilder;
+import com.epam.ta.reportportal.entity.Activity;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.repository.support.PageableExecutionUtils;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Timestamp;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static com.epam.ta.reportportal.dao.util.RecordMappers.ACTIVITY_MAPPER;
+import static com.epam.ta.reportportal.jooq.tables.JActivity.ACTIVITY;
+
+@Repository
+public class ActivityRepositoryCustomImpl implements ActivityRepositoryCustom {
+
+	private static final String PROJECT_ID_COLUMN = "project_id";
+	private static final String CREATION_DATE_COLUMN = "creation_date";
+	private static final String OBJECT_ID_COLUMN = "object_id";
+
+	private DSLContext dsl;
+
+	@Autowired
+	public ActivityRepositoryCustomImpl(DSLContext dsl) {
+		this.dsl = dsl;
+	}
+
+	private final Function<Result<? extends Record>, List<Activity>> ACTIVITY_FETCHER = r -> {
+		Map<Long, Activity> activityMap = Maps.newHashMap();
+		r.forEach(res -> {
+			Long activityId = res.get(ACTIVITY.ID);
+			if (!activityMap.containsKey(activityId)) {
```

Comment:
if we don't have joins that contain duplicates we don't need this logic with fetcher
we can use only mapper everywhere